src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,433 @@
+    private final List<Integer> targets = Arrays.asList(
+        TokenTypes.PARAMETERS,
+        TokenTypes.RECORD_COMPONENTS,
+        TokenTypes.CLASS_DEF,
+        TokenTypes.INTERFACE_DEF
+    );
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode node) {
+        DetailAST ast = getBlockCommentAst().getParent();
+
+        if (ast != null) {
+            if (ast.getType() == TokenTypes.ANNOTATION) {
+                ast = ast.getParent();
+            }
+            if (targetParent.contains(ast.getType())) {
+                DetailAST typeDef = ast.getParent();
+                final DetailAST prevSib = getBlockCommentAst().getParent();
+                if (prevSib.getFirstChild().getType() == TokenTypes.MODIFIERS) {
+                    typeDef = ast;
+                }
+                for (Integer type : targets) {
+                    if (typeDef.getType() == type) {
+                        final List<DetailAST> methodTypeParams =
+                            CheckUtil.getTypeParameters(typeDef);
+                        checkParamOrder(node, new ArrayList<>(), methodTypeParams);
+                    }
+                    if (typeDef.findFirstToken(type) != null) {
+                        final List<DetailAST> methodParams =
+                            getParameters(typeDef.findFirstToken(type));
+                        final List<DetailAST> methodTypeParams =
+                            CheckUtil.getTypeParameters(typeDef);
+                        checkParamOrder(node, methodParams, methodTypeParams);
+                    }
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Make a list consists of elements present in  paramsList and not already exists in the
+     * list itself before and returns hashMap of unique elements from javadoc param list.
+     *
+     * @param javadocParamsList list of javadoc parameters.
+     * @param paramsList list of javadoc parameters.
+     * @return hashMap of unique elements from javadoc param list with integer value.
+     */
+    private static HashMap<String, Integer> helperMethod(List<String> javadocParamsList,
+                                                          List<String> paramsList) {
+        final List<String> javadocIntersectionUniqueTagsList = new ArrayList<>();
+        for (String str : javadocParamsList) {
+            if (paramsList.contains(str) && !javadocIntersectionUniqueTagsList.contains(str)) {
+                javadocIntersectionUniqueTagsList.add(str);
+            }
+        }
+
+        paramsList.removeIf(str -> !javadocIntersectionUniqueTagsList.contains(str));
+        final HashMap<String, Integer> intersection = new HashMap<>();
+        int count = 0;
```

Comment:
why can't we use hashMap?

Reply:
Try to use ConcurrentHashMap. HashMap is not good while multithreading so the use of it might have been restricted due to that.

Reply:
Just curious to know @AkMo3 Is there any module in checkstyle using multithreading?

Most of the modules I came across didn't use any multithreading or anything as such

Reply:
There have been attempts to add this feature in https://github.com/checkstyle/checkstyle/projects/5, but this is not active right now.

Reply:
Interesting one, it will be really great if we use multithreading paradigm used in the modules....checkstyle will run even faster. Handling of threads properly will be a challenge in this project